openstates/nm/legislators.py

```diff
@@ -28,101 +28,74 @@ def scrape_legislator(self, chamber, term, url):
-            properties['party'] = 'Democratic'
-        elif '(R)' in properties['party']:
-            properties['party'] = 'Republican'
-        elif '(DTS)' in properties['party']:
+        if '(D)' in party:
+            party = 'Democratic'
+        elif '(R)' in party:
+            party = 'Republican'
+        elif '(DTS)' in party:
             # decline to state = independent
-            properties['party'] = 'Independent'
+            party = 'Independent'
         else:
             raise Exception("unknown party encountered")
 
-        address = properties.pop('addr_street')
-
-        phone = (properties.pop('office_phone') or
-                 properties.pop('home_phone'))
-
-        leg = Legislator(**properties)
-        leg.add_source(url)
+        photo_url = doc.xpath('//img[@id="ctl00_mainCopy_formViewLegislator_imgLegislator"]/@src')[0]
 
+        email = doc.xpath('//a[@id="ctl00_mainCopy_formViewLegislator_linkEmail"]/text()')
         if email:
-            properties['email'] = email.strip()
+            email = email[0]
         else:
             print "no email"
             email = None
 
-        leg.add_office('district', 'District Address', address=address,
-                       phone=phone, email=email)
+        district_address = ", ".join(doc.xpath(xpath.format('lblAddress')))
+
+        district_phone = doc.xpath(xpath.format('lblHomePhone'))
+        # if they don't have a home phone, check for an office phone
+        if not district_phone:
+            district_phone = doc.xpath(xpath.format('lblOfficePhone'))
+        if district_phone:
+            district_phone = district_phone[0]
+        else:
+            district_phone = None
+
+        capitol_address = doc.xpath(xpath.format('lblCapitolRoom'))
+        if capitol_address:
+            capitol_address = 'Room ' + capitol_address[0] + ' State Capitol, Santa Fe, NM 87501'
+        else:
+            capitol_address = None
+        
+        capitol_phone = doc.xpath(xpath.format('lblCapitolPhone'))
+        if capitol_phone:
+            capitol_phone = "(505) " + capitol_phone[0]
+        else:
+            capitol_phone = None
+
+        leg = Legislator(term=term, chamber=chamber, district=district, full_name=full_name, party=party, photo_url=photo_url)
+        leg.add_source(url)
+
+        leg.add_office('district', 'District Office', address=district_address,
+                        phone=district_phone)
+        leg.add_office('capitol', 'Capitol Office', address=capitol_address,
+                        phone=capitol_phone, email=email)
 
         # committees
         # skip first header row
         for row in doc.xpath('//table[@id="ctl00_mainCopy_gridViewCommittees"]/tr')[1:]:
-            role, committee, note = [x.text_content()
-                                     for x in row.xpath('td')]
+            role, committee, note = [x.text_content() for x in row.xpath('td')]
             committee = committee.title()
```

Comment:
This wasn't being done before, but could you make sure to standardize the committee names by removing "Committee" and "Subcommittee" from them? Also, if there's a way to figure out which `committee` a `subcommittee` is part of, that would be great, too!